modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/LocalSubscription.java

```diff
@@ -98,9 +98,9 @@ public LocalSubscription(OutboundSubscription subscription, String subscriptionI
         // In case of mock subscriptions, this becomes null
         if (null != subscription) {
             //We need to keep the protocolType in basic subscription for notification
-            if(subscription instanceof AMQPLocalSubscription) {
+            if((AMQPUtils.PROTOCOL_TYPE).equals(subscription.getProtocolType())) {
                 setProtocolType(ProtocolType.AMQP);
-            } else if(subscription instanceof MQTTLocalSubscription) {
+            } else if((MQTTUtils.PROTOCOL_TYPE).equals(subscription.getProtocolType())) {
```

Comment:
Logic here is to check if a string is equal to 'AMQP' or 'MQTT' and select the correct protocol type from ProtocolType.AMQP and ProtocolType.MQTT. This could have been done directly using the enums.